src/group/errors.rs

```diff
@@ -4,102 +4,107 @@
-    CannotRemoveSelf = 300,
-    OwnKeyNotFound = 301,
-}
-
-#[derive(PartialEq, Debug)]
-#[repr(u16)]
-pub enum ExporterError {
-    KeyLengthTooLong = 400,
+implement_error! {
+    pub enum GroupError {
+        MLSCiphertextError(MLSCiphertextError) =
+            "See [`MLSCiphertextError`](`crate::framing::errors::MLSCiphertextError`) for details",
+        WelcomeError(WelcomeError) =
+            "See [`WelcomeError`](`WelcomeError`) for details",
+        ApplyCommitError(ApplyCommitError) =
+            "See [`WelcomeError`](`ApplyCommitError`) for details",
+        CreateCommitError(CreateCommitError) =
+            "See [`WelcomeError`](`CreateCommitError`) for details",
+        ConfigError(ConfigError) =
+            "See [`ConfigError`](`crate::config::ConfigError`) for details",
+        ExporterError(ExporterError) =
+            "See [`ExporterError`](`ExporterError`) for details",
+        SecretTypeError(SecretTypeError) =
+            "See [`SecretTypeError`](`crate::tree::secret_tree::SecretTypeError`) for details",
+    }
 }
 
-impl From<MLSCiphertextError> for GroupError {
-    fn from(_e: MLSCiphertextError) -> Self {
-        GroupError::DecryptionError
+implement_error! {
+    pub enum WelcomeError {
+        CiphersuiteMismatch =
+            "Ciphersuites in the Welcome message and the corresponding key package bundle don't match.",
+        JoinerSecretNotFound =
+            "No joiner secret found the Welcome message.",
+        MissingRatchetTree =
+            "No ratchet tree available to build initial tree after receiving a Welcome message",
+        TreeHashMismatch =
+            "The tree hash computed does not match the one in the GroupInfo.",
+        ConfirmationTagMismatch =
+             "The computed confirmation tag does not match the expected one.",
+        InvalidRatchetTree =
+            "Invalid ratchet tree in Welcome message.",
+        InvalidGroupInfoSignature =
+            "The signature on the GroupInfo is not valid.",
+        GroupInfoDecryptionFailure =
+            "Unable to decrypt the GroupInfo",
+        DuplicateRatchetTreeExtension =
+            "Found more than one ratchet tree extension in the Welcome message.",
+        UnsupportedMlsVersion =
+            "The Welcome message uses an unsupported MLS version",
+        UnknownError =
+            "An unknown error occurred.",
     }
 }
 
-impl From<TreeError> for WelcomeError {
-    fn from(e: TreeError) -> WelcomeError {
-        match e {
-            TreeError::DuplicateIndex => WelcomeError::InvalidRatchetTree,
-            TreeError::InvalidArguments => WelcomeError::InvalidRatchetTree,
-            TreeError::InvalidUpdatePath => WelcomeError::InvalidRatchetTree,
-            TreeError::UnknownError => WelcomeError::UnknownError,
-        }
+implement_error! {
+    pub enum ApplyCommitError {
+        EpochMismatch =
+            "Couldn't apply commit. The epoch of the group context and MLSPlaintext didn't match.",
```

Comment:
Spelling: If there is an actual `struct` of the same name, it should be capitalized the same way.
commit -> Commit